odoo/orm/fields_relational.py

```diff
@@ -568,11 +576,18 @@ def convert_to_display_name(self, value, record):
 
     def get_depends(self, model):
         depends, depends_context = super().get_depends(model)
-        if not self.compute and isinstance(self.domain, list):
+        if not self.compute:
+            try:
+                domain = self.get_comodel_domain(model)
+            except ValueError:
+                if model.env.registry.ready:
+                    raise
+                # "external ID not found" is raised during installation for callable domains
+                # ignore the domain in that case
+                domain = Domain.TRUE
             depends = unique(itertools.chain(depends, (
-                self.name + '.' + arg[0]
-                for arg in self.domain
-                if isinstance(arg, (tuple, list)) and isinstance(arg[0], str)
+                self.name + '.' + condition.field_expr
+                for condition in domain.iter_conditions()
             )))
         return depends, depends_context
```

Comment:
Actually I don't understand the fix.

We already update one2many inverses when we write/create on a many2one, see `_update_inverses` methods and its usages (and it takes care of the domain), then we shouldn't to invalidate one2many in `modified` at the root level (It is already the case if a domain exists on the one2many and it actually already generate extra invalidation for no reason...).

Them, changing `get_depends` is only useful for the `_description_depends` (not sure if this method is still useful).